core-rust/state-manager/src/transaction/executable_logic.rs

```diff
@@ -64,43 +63,65 @@ impl ExecutionConfigurator {
     ) -> Self {
         Self {
             scrypto_vm: ScryptoVm::<DefaultWasmEngine>::default(),
-            costing_parameters,
             execution_configs: HashMap::from([
                 (
                     ConfigType::Genesis,
                     ExecutionConfig::for_genesis_transaction(network.clone())
+                        .with_costing_parameters(costing_parameters)
                         .with_kernel_trace(engine_trace),
                 ),
                 (
                     ConfigType::OtherSystem,
-                    ExecutionConfig {
-                        max_number_of_events: 1_000_000,
-                        ..ExecutionConfig::for_system_transaction(network.clone())
-                            .with_kernel_trace(engine_trace)
-                    },
+                    ExecutionConfig::for_system_transaction(network.clone())
+                        .with_costing_parameters(costing_parameters)
+                        .with_limit_parameters(LimitParameters {
```

Comment:
`for_system_transaction` turns off the limit module and costing module now, so I think we can remove this override.